kedro/templates/project/hooks/utils.py

```diff
@@ -0,0 +1,160 @@
+lint_pyproject_requirements = """
+[tool.ruff]
+select = [
+    "F",  # Pyflakes
+    "E",  # Pycodestyle
+    "W",  # Pycodestyle
+    "UP",  # pyupgrade
+    "I",  # isort
+    "PL", # Pylint
+]
+ignore = ["E501"]  # Black takes care of line-too-long
+"""
+
+test_requirements = "pytest-cov~=3.0\npytest-mock>=1.7.1, <2.0\npytest~=7.2"
+test_pyproject_requirements = """
+[tool.pytest.ini_options]
+addopts = \"\"\"
+--cov-report term-missing \\
+--cov src/{{ cookiecutter.python_package }} -ra
+\"\"\"
+
+[tool.coverage.report]
+fail_under = 0
+show_missing = true
+exclude_lines = ["pragma: no cover", "raise NotImplementedError"]
+"""
+
+docs_pyproject_requirements = """
+docs = [
+    "docutils<0.18.0",
+    "sphinx~=3.4.3",
+    "sphinx_rtd_theme==0.5.1",
+    "nbsphinx==0.8.1",
+    "nbstripout~=0.4",
+    "sphinx-autodoc-typehints==1.11.1",
+    "sphinx_copybutton==0.3.1",
+    "ipykernel>=5.3, <7.0",
+    "Jinja2<3.1.0",
+    "myst-parser~=0.17.2",
+]
+"""
+
+def _validate_range(start, end):
+    if int(start) > int(end):
+        message = f"'{start}-{end}' is an invalid range for project add-ons."
+        click.secho(message, fg="red", err=True)
+        sys.exit(1)
+
+def _validate_selection(add_ons):
+    for add_on in add_ons:
+        if int(add_on) < 1 or int(add_on) > 5:
+            message = f"'{add_on}' is not a valid selection."
+            click.secho(message, fg="red", err=True)
+            sys.exit(1)
+
+
+def parse_add_ons_input(add_ons_str):
+    """Parse the add-ons input string.
+
+    Args:
+        add_ons_str: Input string from prompts.yml.
+
+    Returns:
+        list: List of selected add-ons as strings.
+    """
+    if add_ons_str == "all":
+        return ["1", "2", "3", "4", "5"]
+    if add_ons_str == "none":
+        return []
+
+    # Split by comma
+    add_ons_choices = add_ons_str.split(",")
+    selected = []
+
+    for choice in add_ons_choices:
+        if "-" in choice:
+            start, end = choice.split("-")
+            _validate_range(start, end)
+            selected.extend(str(i) for i in range(int(start), int(end) + 1))
+        else:
+            selected.append(choice.strip())
+
+    _validate_selection(selected)
+    return selected
+
+
+def setup_template_add_ons(selected_add_ons_list, requirements_file_path, pyproject_file_path, pyproject_src_file_path):
+    """Removes directories and files related to unwanted addons from
+    a Kedro project template. Adds the necessary requirements for
+    the addons that were selected.
+
+    Args:
+        selected_add_ons_list: a list containing numbers from 1 to 5,
+        representing specific add-ons.
+        requirements_file_path: the path to the requirements.txt file.
+        pyproject_file_path: the path to the pyproject.toml file
+        located on the the root of the template.
+        pyproject_src_file_path: the path to the pyproject.toml file
+        located inside the `src` directory.
```

Comment:
Nit: the formatting looks a bit weird for this doc string. Could you add some tabs to make it easier to read what are the arguments and what the description?